plugins/modules/ec2_vpc_route_table.py

```diff
@@ -650,81 +650,81 @@ def associate_gateway(connection, module, route_table, gateway_id):
+        except AnsibleEC2Error as e:
+            module.fail_json_aws_error(e)
 
     return changed
 
 
-def ensure_route_table_absent(connection, module):
+def ensure_route_table_absent(connection, module: AnsibleAWSModule) -> Dict[str, bool]:
     lookup = module.params.get("lookup")
     route_table_id = module.params.get("route_table_id")
     tags = module.params.get("tags")
     vpc_id = module.params.get("vpc_id")
     purge_subnets = module.params.get("purge_subnets")
+    route_table = None
 
-    if lookup == "tag":
-        if tags is not None:
-            route_table = get_route_table_by_tags(connection, module, vpc_id, tags)
-        else:
-            route_table = None
+    if lookup == "tag" and tags is not None:
+        route_table = get_route_table_by_tags(connection, module, vpc_id, tags)
     elif lookup == "id":
         route_table = get_route_table_by_id(connection, module, route_table_id)
 
     if route_table is None:
         return {"changed": False}
 
     # disassociate subnets and gateway before deleting route table
-    if not module.check_mode:
-        ensure_subnet_associations(
+    changed = False
+    if module.check_mode:
+        changed = True
+    else:
+        changed = ensure_subnet_associations(
             connection=connection, module=module, route_table=route_table, subnets=[], purge_subnets=purge_subnets
         )
-        disassociate_gateway(connection=connection, module=module, route_table=route_table)
+        changed |= disassociate_gateway(connection=connection, module=module, route_table=route_table)
         try:
-            connection.delete_route_table(aws_retry=True, RouteTableId=route_table["RouteTableId"])
-        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-            module.fail_json_aws(e, msg="Error deleting route table")
+            changed |= delete_route_table(connection, route_table_id=route_table["RouteTableId"])
+        except AnsibleEC2Error as e:
+            module.fail_json_aws_error(e)
 
-    return {"changed": True}
+    return {"changed": changed}
 
 
-def get_route_table_info(connection, module, route_table):
-    result = get_route_table_by_id(connection, module, route_table["RouteTableId"])
-    try:
-        result["Tags"] = describe_ec2_tags(connection, module, route_table["RouteTableId"])
-    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-        module.fail_json_aws(e, msg="Couldn't get tags for route table")
-    result = camel_dict_to_snake_dict(result, ignore_list=["Tags"])
-    # backwards compatibility
-    result["id"] = result["route_table_id"]
+def get_route_table_info(connection, module: AnsibleAWSModule, route_table: Dict[str, Any]) -> Optional[Dict[str, Any]]:
+    result = get_route_table_by_id(connection, module, route_table["RouteTableId"]) or {}
+    if result:
+        try:
```

Comment:
`describe_ec2_tags()` already handles the botocore exceptions so I don't think the try/catch is necessary here.

Reply:
done!!